addons/test/cert-test.c

```diff
@@ -11,6 +11,26 @@ struct PackedStruct {
     short b;
 };
 
+void api01()
+{
+    const size_t String_Size = 20;
+    struct bad_node_s
+    {
+        char name[String_Size];
+        struct bad_node_s* next;      // cert-API01-C
```

Comment:
What is the definition for "sensitive data"?

For me "sensitive data" would be for instance a password.

A pointer is not "sensitive" but rather "vulnerable". The pointer address does not mean anything to humans so it's not the "primary" information you'd want to learn as an attacker. But if the attacker can arbitrarily read/write a pointer in memory that would be terrible from a security perspective.

So well.. basically I think it is unintuitive to say that pointers are "sensitive". So you should not warn about pointers.

But on the other hand, I can see that it could be extra vulnerable with pointers after arrays. So should you only warn about pointers?

I would like to read about this rule on CERT but I do not find this information.

Reply:
I send you the link to the CERT rule. Basically "sensitive data" means any other data in that structure. So it's actually an overflow check for structure definitions.
From a security point of view you can produce a crash or inject code by overwriting other structure members just by initializing the structure.

Reply:
Sorry for late reply... if all data is sensitive then putting an array after the array should be a warning also?